### PR TITLE
fix(odm): offset and limit params not being passed to the search pipeline

### LINKED ISSUE #271: offset parameter is ignored in search endpoint
### Please explain the bug.

`offset` parameter is not working for `/search` endpoint as no matter the number, it always returns the first/most recent page

### How can I reproduce it?

```
curl -X 'POST' \
  'https://api.saebasol.org/api/hitomi/search' \
  -H 'accept: */*' \
  -H 'Content-Type: application/json' \
  -d '{
  "offset": 1,
  "query": [
    "female:loli"
  ]
}'
```
returns same `result` as 
```
curl -X 'POST' \
  'https://api.saebasol.org/api/hitomi/search' \
  -H 'accept: */*' \
  -H 'Content-Type: application/json' \
  -d '{
  "offset": 2,
  "query": [
    "female:loli"
  ]
}'
```

### Logs/tracebacks

_No response_

### Heliotrope version or Docker image version

```console
6.4.3 https://api.saebasol.org/docs
```

### Python Version

_No response_

### OS

_No response_

### Additional Description

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by FluffyDiscord**
- [x] Changes to this PR can be deployed immediately.

https://github.com/Saebasol/Heliotrope/issues/271

### CODE REVIEW

**Review by SaidBySolo — COMMENTED** [review]
All arguments to this function must be required.

https://github.com/Saebasol/Heliotrope/blob/d99e7cd7152b9f1838b29394b3c4166abad20790/heliotrope/database/odm/__init__.py#L113

And can you help me with the [commit convention?](https://www.conventionalcommits.org/en/v1.0.0/)

**Comment by FluffyDiscord** [comment]
> All arguments to this function must be required.
> 
> https://github.com/Saebasol/Heliotrope/blob/d99e7cd7152b9f1838b29394b3c4166abad20790/heliotrope/database/odm/__init__.py#L113
> 
> And can you help me with the [commit convention?](https://www.conventionalcommits.org/en/v1.0.0/)

would this do ?

File: heliotrope/database/odm/__init__.py [hunk 1]
```diff
@@ -150,15 +150,15 @@ async def search(
...
+        count_pipeline, pipeline = self.make_search_pipeline(title, query, offset, limit)
 
         try:
             count = await self.collection.aggregate(count_pipeline).next()
         except StopAsyncIteration:
             results: list[HitomiInfoJSON] = []
             count = 0
         else:
-            results = await self.collection.aggregate(pipeline).to_list(15)
+            results = await self.collection.aggregate(pipeline).to_list(limit)
```
Comment (SaidBySolo) [inline heliotrope/database/odm/__init__.py]:
```suggestion
            results = await self.collection.aggregate(pipeline).to_list(15)
```

See Line 151

File: heliotrope/database/odm/__init__.py [hunk 2]
```diff
@@ -150,15 +150,15 @@ async def search(
 
         offset = offset * limit
         title, query = self.parse_query(querys)
-        count_pipeline, pipeline = self.make_search_pipeline(title, query)
+        count_pipeline, pipeline = self.make_search_pipeline(title, query, offset, limit)
```
Comment (SaidBySolo) [inline heliotrope/database/odm/__init__.py]:
```suggestion
        count_pipeline, pipeline = self.make_search_pipeline(
            title, query, offset, limit
        )
```

Code style

**Review by SaidBySolo — CHANGES_REQUESTED** [review]

File: heliotrope/database/odm/__init__.py (hunk 1 above)
Comment (FluffyDiscord) [reply heliotrope/database/odm/__init__.py]: my bad, shoud not touch things randomly

**Review by SaidBySolo — APPROVED** [review]
Thank you so much for your contribution!

**Comment by FluffyDiscord** [comment]
> Thank you so much for your contribution!

Thank you for being patient with me

**Pull request merged**